src/pages/signin.tsx

```diff
@@ -0,0 +1,105 @@
+import { useAuth } from "components/AuthProvider";
+import { UserStatus } from "components/UserStatus";
+import { useRouter } from "next/router";
+import React, { useEffect, useRef, useState } from "react";
+
+const defaultEmail = "admin@example.com";
+const defaultPassword = "admin123";
+
+export default function SignIn() {
+  const { auth, initializing, getRedirect, clearRedirect, user, error } =
+    useAuth();
+  const [email, setEmail] = useState<string>(defaultEmail);
+  const [pswd, setPswd] = useState<string>(defaultPassword);
+  const [signInInProgress, setInProgress] = useState(false);
+  const mounted = useRef<boolean>();
+  const router = useRouter();
+
+  /* Guard if page is navigated away while sign in process is still active */
+  useEffect(() => {
+    mounted.current = true;
+
+    return () => {
+      mounted.current = false;
+    };
+  }, []);
+
+  useEffect(() => {
+    if (!initializing) {
+      if (user) {
+        const redirect = getRedirect();
+        console.log("redirect: ", redirect);
+        if (redirect !== null) {
+          router.push(redirect); // go to page which redirected to login
+          clearRedirect();
+        } else {
+          router.back(); // go to previous page
+        }
+      }
+    }
+  }, [router, getRedirect, clearRedirect, initializing, user]);
+
+  const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    if (email && pswd) {
+      try {
+        setInProgress(true);
+        await auth.signIn(email, pswd, 2000);
```

Comment:
Sorry oh but thinking about it again we might want to consider using next-auth. That way we can put the email and password on the server side. 

I think it would make managing the "authenticated" state easier

Reply:
We agreed to bootstrap sth then improve on it
Thats what I did

I will checkout the next auth again